### DISCUSSION TITLE
Renovate consuming increasingly more RAM with execution?

### DISCUSSION THREAD

**Discussion by ezloj** [renovatebot/renovate#29454] [Request Help] [answered]
### What would you like help with?

I think I found a bug

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitLab

### Please tell us more about your question or problem

Renovate over time will consume more and more RAM. Because I'm running it in k8s gitlab-runner pod, the pod has limited RAM. During renovate run, once RAM usage gets close to the pod's limit (about 20 minutes after execution begins), the renovate run fails because it's child node process to update node dependncies gets killed due to OOM. This doesn't produce a pod OOM - just the node process dying like this:

```
/usr/local/bin/node: line 6:    36 Killed                  /opt/containerbase/tools/node/18.20.3/bin/node --use-openssl-ca "$@"
```

### Logs (if relevant)

<details><summary>Logs</summary>

```

/usr/local/bin/node: line 6:    36 Killed                  /opt/containerbase/tools/node/18.20.3/bin/node --use-openssl-ca "$@"

```

</details>

**Comment by rarkins** [comment]
RAM use, and increasing it over time, is not automatically a bug. We need more details before deciding if this is something worth optimizing.

**Reply by rarkins** [reply]
Was this your first time using renovate, or is this only a recent problem? Because this was also raised today: https://github.com/renovatebot/renovate/discussions/29448

**Reply by ezloj** [reply]
I'm running the 37 version of official docker image. First time seeing this issue. First occurrence 2 days ago. Steady renovate user for 4 years here

**Reply by rarkins** [reply]
In that case let's see if the re2 revert for the linked discussion fixes your problem

**Reply by ezloj** [reply]
Thank you. Instant response as usual. Best maintainer ever 🙌

**Reply by rarkins** [reply, marked as answer]
FYI we've done a check of our hosted app and found a negligible increase in ram usage when we upgraded to `37.388.1`, which lowers my expectations that `re2` was the culprit and the revert of its upgrade will make your problems go away. It may have been coincidental. But let's see

**Reply by ezloj** [reply]
Looks like my nightly renovate run on latest version 37 image is okay, so problem solved

**Reply by rarkins** [reply]
Thanks for the feedback!
